envoy/http/async_client.h

```diff
@@ -349,6 +401,18 @@ class AsyncClient {
   virtual Request* send(RequestMessagePtr&& request, Callbacks& callbacks,
                         const RequestOptions& options) PURE;
 
+  /**
+   * Starts a new request asynchronously with the given headers.
```

Comment:
would be useful to document the difference of this approach vs start below for a caller